FreeRTOS-Plus/Demo/AWS/Device_Shadow_Windows_Simulator/Device_Shadow_Demo/DemoTasks/ShadowDemoMainExample.c

```diff
@@ -607,205 +807,291 @@ void prvShadowDemoTask( void * pvParameters )
 
     /****************************** Connect. ******************************/
 
-    demoStatus = xEstablishMqttSession( &xMqttContext,
-                                        &xNetworkContext,
-                                        &xBuffer,
-                                        prvEventCallback );
-
-    if( demoStatus == pdFAIL )
+    /* This demo runs a single loop unless there are failures in the demo execution.
+     * In case of failures in the demo execution, demo loop will be retried for up to
+     * SHADOW_MAX_DEMO_LOOP_COUNT times. */
+    do
     {
-        /* Log error to indicate connection failure. */
-        LogError( ( "Failed to connect to MQTT broker." ) );
-    }
+        /****************************** Connect. ******************************/
 
-    if( demoStatus == pdPASS )
-    {
-        /* Try to delete any Shadow document in the cloud. This is done to
-         * ensure possbile previous Shadow documents do not affect the state
-         * expected in the current demo. */
-        demoStatus = xPublishToTopic( &xMqttContext,
-                                      SHADOW_TOPIC_STRING_DELETE( democonfigTHING_NAME ),
-                                      SHADOW_TOPIC_LENGTH_DELETE( THING_NAME_LENGTH ),
-                                      pcUpdateDocument,
-                                      0U );
-    }
+        xDemoStatus = xEstablishMqttSession( &xMqttContext,
+                                             &xNetworkContext,
+                                             &xBuffer,
+                                             prvEventCallback );
 
-    /********************* Subscribe to Shadow topics. ************************/
+        if( xDemoStatus == pdFAIL )
+        {
+            /* Log error to indicate connection failure. */
+            LogError( ( "Failed to connect to MQTT broker." ) );
+        }
 
-    /* Then try to subscribe the Shadow topics. */
+        /**************** Attempt to delete Shadow document. ******************/
 
-    if( demoStatus == pdPASS )
-    {
-        demoStatus = xSubscribeToTopic( &xMqttContext,
-                                        SHADOW_TOPIC_STRING_UPDATE_DELTA( democonfigTHING_NAME ),
-                                        SHADOW_TOPIC_LENGTH_UPDATE_DELTA( THING_NAME_LENGTH ) );
-    }
+        if( xDemoStatus == pdPASS )
+        {
+            /* Reset the shadow delete status flags. */
+            xDeleteResponseReceived = pdFALSE;
+            xShadowDeleted = pdFALSE;
+
+            /* First of all, try to delete any Shadow document in the cloud.
+             * Try to subscribe to `/delete/accepted` and `/delete/rejected` topics. */
+            xDemoStatus = xSubscribeToTopic( &xMqttContext,
```

Comment:
I think subscribing and unsubscribing is unnecessary.  There is an IoT Core feature such that responses to the API calls are always delivered regardless of subscription.

Reply:
Thanks for the suggestion. As discussed offline, this can be considered as a follow up change after analyzing the feature.

Reply:
Verified that the feature support is not available for Shadow.